recommenders/models/sasrec/model_ssept.py

```diff
@@ -0,0 +1,522 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import tensorflow as tf
+import numpy as np
+import sys
+
+
+class MultiHeadAttention(tf.keras.layers.Layer):
+    """
+    - Q (query), K (key) and V (value) are split into multiple heads (num_heads)
+    - each tuple (q, k, v) are fed to scaled_dot_product_attention
+    - all attention outputs are concatenated
+    """
+
+    def __init__(self, attention_dim, num_heads, dropout_rate):
+        super(MultiHeadAttention, self).__init__()
+        self.num_heads = num_heads
+        self.attention_dim = attention_dim
+        assert attention_dim % self.num_heads == 0
+        self.dropout_rate = dropout_rate
+
+        self.depth = attention_dim // self.num_heads
```

Comment:
This code for multihead attention is the same as in the other file? If so, it would be good to refactor to not repeat code